app.js

```diff
@@ -1,12 +1,149 @@
+    linkLive: 'https://sebastiansolisfenger.github.io/my-portfolio-microverse/',
+    linkSource:
+      'https://github.com/SebastianSolisFenger/my-portfolio-microverse',
+  },
+  {
+    name: 'Multi-Post Stories',
+    details: ['FACEBOOK', '•', 'Back End Dev', '•', '2015'],
+    smallDescription:
+      'Experimental content creation feature that allows users to add to an existing story over the course of a day without spamming their friends.',
+    bigDescription:
+      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent",
+    imageUrl: '/img/SnapshootPortfolio2.png',
+    technologies: ['html', 'Ruby on rails', 'css', 'javaScript'],
+    linkLive: 'https://sebastiansolisfenger.github.io/my-portfolio-microverse/',
+    linkSource:
+      'https://github.com/SebastianSolisFenger/my-portfolio-microverse',
+  },
+  {
+    name: 'Facebook 360',
+    details: ['FACEBOOK', '•', 'Back End Dev', '•', '2015'],
+    smallDescription:
+      "Exploring the future of media in Facebook's first Virtual Reality app; a place to discover and enjoy 360 photos and videos on Gear VR.",
+    bigDescription:
+      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent",
+    imageUrl: 'img/popup_big.png',
+    technologies: ['html', 'Ruby on rails', 'css', 'javaScript'],
+    linkLive: 'https://sebastiansolisfenger.github.io/my-portfolio-microverse/',
+    linkSource:
+      'https://github.com/SebastianSolisFenger/my-portfolio-microverse',
+  },
+  {
+    name: 'Uber navigation',
+    details: ['Uber', '•', 'Lead developer', '•', '2018'],
+    smallDescription:
+      "Exploring the future of media in Facebook's first Virtual Reality app; a place to discover and enjoy 360 photos and videos on Gear VR.",
+    bigDescription:
+      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent",
+    imageUrl: 'img/popup_big.png',
+    technologies: ['html', 'Ruby on rails', 'css', 'javaScript'],
+    linkLive: 'https://sebastiansolisfenger.github.io/my-portfolio-microverse/',
+    linkSource:
+      'https://github.com/SebastianSolisFenger/my-portfolio-microverse',
+  },
+];
```

Comment:
[REQUIRED] Nice job on storing all project-related data as `objects` within an `array` as it is required by this milestone :clap:. Now let's use this `data` to create all our project cards dynamically, right now I can see in your [index.html](https://github.com/SebastianSolisFenger/my-portfolio-microverse/blob/proto_POP-UP/index.html) the project cards are still being created statically. Please refer to the below code lines:
https://github.com/SebastianSolisFenger/my-portfolio-microverse/blob/b0dc5eb0c6e57e4cd8e824796df62e467b5fc279/index.html#L81-L386

To answer the question of why do we need to create cards dynamically, let's imagine you want to add a new project to your portfolio, with `static` code, you need to create another  `card ` in your HTML file and put all the data manually, meaning that you have to repeat yourself every time you want to add a new project, that is not good practice in terms of keeping your code `DRY`. `Dynamic code`, on the other hand, allows you to do it by just adding a new object to your `projects array` and the rest will be taken care of with the functions you create in your `script`. On top of that, this is one of the main requirements of this milestone, so let's create our project cards `dynamically`.

**_hints:_**

**Step-1**: Make sure the values of the project `objects` within the `array list` are correct.

**Step-2**: Select the main container for your cards, I think, it is section class `works`.

**Step-3**: Now we can iterate through our `array list`. You have many options like `for loop`, `forEach`, `map` and etc. You can go with either option depending on your convenience. I am gonna use `map` for this particular case. So we are looping the `array` each `element` is an object, in order to get the value from the object you can specify the appropriate key, something like `element.name`. Now keeping the above information in mind, we need to create an `html` element (project card) in each iteration and `append` it to our main container (mentioned in step-2).

_sample code:_
```JAVASCRIPT
const mainProjectsContainer = document.querySelector('.works');

mainProjectsContainer.innerHTML += myProjects.map((project) => {
return ( 
` <div class="projectCard">
        <img src=${project.image} alt="project screenshot">
        <div class="project-description">
          <h3 class="project-title">${project.title}</h3>
          <ul>
          // you can store your technologies as an array list within your object, when it is case, you can use the same logic to create `<li/>` tags dynamically
          </ul>
          <button type="button" class="see-project-btn">See this porject</button>
        </div>
      </div>
    </div>
  `)
}).join('');
```
**Step-4**: Delete your static `project cards` from your [index.html](https://github.com/joel-kalema/Portfolio/blob/popup/index.html). I hope it is working as expected :pray: .

_note:_  You can use the same logic with any other `loop`. Please note that the above code is not a direct solution, it is just a sample code to demonstrate the above-mentioned steps.

Reply:
Hi @Hope1226.. I started all over again with another of my templates because I did not which one would you guys were waiting for. So, is it this what you are making reference to?

- ![image](https://user-images.githubusercontent.com/88522494/161825119-e5a8d9c3-1b6f-4c61-a390-f038b4a4c14a.png)

- ![image](https://user-images.githubusercontent.com/88522494/161825208-3d5374c9-890e-4619-9a94-1f6347032bb0.png)

- ![image](https://user-images.githubusercontent.com/88522494/161825257-b5c39396-2d15-41c8-aab3-fdd63685350c.png)

Reply:
I guess that's what you mean by project cards dynamically, right? @Hope1226

Reply:
Hi @SebastianSolisFenger, as long as the above code is creating the project cards dynamically, you are good to go with that one.

Reply:
Yes, it is working, but I still need to fix some style stuff, just wanted to make sure that was what you meant with the js file. @Hope1226

Reply:
![image](https://user-images.githubusercontent.com/88522494/161826930-9d3c67ec-97d9-403c-a980-36f39395d27f.png)

Reply:
![image](https://user-images.githubusercontent.com/88522494/161827037-bd11b85d-158c-4eb2-a282-b80658b21421.png)


AS u can see there it is just the div, @Hope1226

Reply:
@SebastianSolisFenger , nice :blush:  that is exactly what we want here, you can ask for another review, please kindly make sure the `popup window` is functioning as expected. cheers!

Reply:
hey @Hope1226 , I've been working a lot today, taking into account I started kinda from scratch again. Do I have time to submit it tomorrow?

Reply:
@Hope1226